puppet/modules/slave/manifests/rbenv.pp

```diff
@@ -0,0 +1,21 @@
+# rbenv Configuration
+# @api private
+class slave::rbenv {
+  file { '/home/jenkins/.rbenv':
+    ensure  => directory,
+    owner   => 'jenkins',
+    group   => 'jenkins',
+    require => User['jenkins'],
+  }
+
+  class { 'rbenv':
+    install_dir => '/home/jenkins/.rbenv',
+    require     => [File['/home/jenkins/.rbenv'], Package['gcc-c++']],
+  }
+
+  ensure_packages(['gcc-c++'])
+
+  rbenv::build { '3.1.0': }
+  rbenv::build { '3.0.4': }
+  rbenv::build { '2.7.6': }
```

Comment:
These are not the most up to date versions of Ruby but our choices are dictated by what's available from ruby-build.

Reply:
"from ruby-build in EPEL", which we can influence, if we want.

Looking at https://canidepend.com/#ruby, we'd want 2.7.8, 3.0.4, 3.1.2 to match EL?